src/ex2/CompileVisitor.java

```diff
@@ -11,286 +11,329 @@
+		}
+	}
+
+	@Override
+	public void visit(IfStatement ifStatement) {
+		ifStatement.cond().accept(this);
+		addLine("br i1 %_" + (lastRegisterNumber - 1) + ", " + "label %label" + lastLabelNumber + ", label %label"
+				+ (lastLabelNumber + 1));
+		if (!(ifStatement.thencase() instanceof BlockStatement)) {
+			builder.append("label" + lastLabelNumber++ + ":\n");
+			ifStatement.thencase().accept(this);
+		} else {
+			ifStatement.thencase().accept(this);
+		}
+		addLine("br label %label" + (lastLabelNumber + 1) + ":\n");
+		if (!(ifStatement.elsecase() instanceof BlockStatement)) {
+			builder.append("label" + lastLabelNumber++ + ":\n");
+			ifStatement.elsecase().accept(this);
+		} else {
+			ifStatement.elsecase().accept(this);
+		}
+		addLine("br label %label" + lastLabelNumber + ":\n");
+		builder.append("label" + lastLabelNumber++ + ":\n");
+	}
+
+	@Override
+	public void visit(WhileStatement whileStatement) {
+
+	}
+
+	// ARRAYS
+	@Override
+	public void visit(AssignArrayStatement assignArrayStatement) {
+
+	}
+
+	@Override
+	public void visit(ArrayAccessExpr e) {
+	}
+
+	@Override
+	public void visit(ArrayLengthExpr e) {
+
+	}
+
+	@Override
+	public void visit(NewIntArrayExpr e) {
+
+	}
+
+	@Override
+	public void visit(SysoutStatement sysoutStatement) {
+		sysoutStatement.arg().accept(this);
+		addLine("call void (i32) @print_int(i32 " + expr + ")");
+	}
+
+	@Override
+	public void visit(AssignStatement assignStatement) {
+		String type = TypeDecider.llvmType(assignStatement.getSymbolTable().varLookup(assignStatement.lv()).getDecl());
+		assignStatement.rv().accept(this);
+		addLine("store " + type + " " + expr + ", " + type + "* " + "%" + assignStatement.lv());
+	}
+
+	// Exp and trees
+	@Override
+	public void visit(AndExpr e) {
+		// implement short circuit
+	}
+
+	@Override
+	public void visit(LtExpr e) {
+		String lv;
+		e.e1().accept(this);
+		lv = expr;
+		e.e2().accept(this);
+		addLine("%_" + lastRegisterNumber++ + " = icmp slt i32 " + lv + ", " + expr);
+	}
+
+	private void visitBinaryExp(BinaryExpr e, String opt) {
+		String e1, e2;
+		e.e1().accept(this);
+		e1 = expr;
+		e.e2().accept(this);
+		e2 = expr;
+		expr = "%_" + lastRegisterNumber;
+		addLine("%_" + lastRegisterNumber++ + " = " + opt + " i32 " + e1 + ", " + e2);
+	}
+
+	@Override
+	public void visit(AddExpr e) {
+		visitBinaryExp(e, "add");
+	}
+
+	@Override
+	public void visit(SubtractExpr e) {
+		visitBinaryExp(e, "sub");
+	}
+
+	@Override
+	public void visit(MultExpr e) {
+		visitBinaryExp(e, "mul");
+	}
+
+	@Override
+	public void visit(NotExpr e) {
+		e.accept(this);
+		addLine("%_" + lastRegisterNumber + " = xor i1 " + expr + ", 1");
+		expr = "%_" + lastRegisterNumber++;
+	}
+
+	@Override
+	public void visit(IntegerLiteralExpr e) {
+		expr = e.num() + "";
+	}
+
+	@Override
+	public void visit(TrueExpr e) {
+		expr = "1";
+	}
+
+	@Override
+	public void visit(FalseExpr e) {
+		expr = "0";
+	}
+
+	@Override
+	public void visit(IdentifierExpr e) {
```

Comment:
should handle a case it's a field variable, also you always load vars as int

Reply:
you're right, i'll update it